tests/generators/kzg_4844/main.py

```diff
@@ -0,0 +1,580 @@
+    expect_exception(spec.verify_blob_kzg_proof_batch, VALID_BLOBS, commitments, commitments_invalid_notCurve)
+    yield 'verify_blob_kzg_proof_batch_case_not_on_curve', {
+        'input': {
+            'blob': encode_hex_list(VALID_BLOBS),
+            'commitment': encode_hex_list(commitments_invalid_notCurve),
+            'proof': encode_hex_list(proofs),
+        },
+        'output': None
+    }
+
+    # Edge case: Invalid blob
+    blobs_invalid = VALID_BLOBS[:4] + [BLOB_INVALID] + VALID_BLOBS[5:]
+    expect_exception(spec.verify_blob_kzg_proof_batch, blobs_invalid, commitments, proofs)
+    yield 'verify_blob_kzg_proof_batch_case_invalid_blob', {
+        'input': {
+            'blob': encode_hex_list(blobs_invalid),
+            'commitment': encode_hex_list(commitments),
+            'proof': encode_hex_list(proofs),
+        },
+        'output': None
+    }
+
+    # Edge case: Blob length different
+    expect_exception(spec.verify_blob_kzg_proof_batch, VALID_BLOBS[:-1], commitments, proofs)
+    yield 'verify_blob_kzg_proof_batch_case_blob_length_different', {
+        'input': {
+            'blob': encode_hex_list(VALID_BLOBS[:-1]),
+            'commitment': encode_hex_list(commitments),
+            'proof': encode_hex_list(proofs),
+        },
+        'output': None
+    }
+
+    # Edge case: Commitment length different
+    expect_exception(spec.verify_blob_kzg_proof_batch, VALID_BLOBS, commitments[:-1], proofs)
+    yield 'verify_blob_kzg_proof_batch_case_commitment_length_different', {
+        'input': {
+            'blob': encode_hex_list(VALID_BLOBS),
+            'commitment': encode_hex_list(commitments[:-1]),
+            'proof': encode_hex_list(proofs),
+        },
+        'output': None
+    }
+
+    # Edge case: Proof length different
+    expect_exception(spec.verify_blob_kzg_proof_batch, VALID_BLOBS, commitments, proofs[:-1])
+    yield 'verify_blob_kzg_proof_batch_case_proof_length_different', {
+        'input': {
+            'blob': encode_hex_list(VALID_BLOBS),
+            'commitment': encode_hex_list(commitments),
+            'proof': encode_hex_list(proofs[:-1]),
+        },
+        'output': None
+    }
+
+
+def create_provider(fork_name: SpecForkName,
+                    handler_name: str,
+                    test_case_fn: Callable[[], Iterable[Tuple[str, Dict[str, Any]]]]) -> gen_typing.TestProvider:
+
+    def prepare_fn() -> None:
+        # Nothing to load / change in spec. Maybe in future forks.
+        # Put the tests into the general config category, to not require any particular configuration.
+        return
+
+    def cases_fn() -> Iterable[gen_typing.TestCase]:
+        for data in test_case_fn():
+            (case_name, case_content) = data
+            yield gen_typing.TestCase(
+                fork_name=fork_name,
+                preset_name='general',
+                runner_name='bls',
```

Comment:
Given that we plan to deprecate the BLS signature test vectors in CL specs, to avoid confusion, it's better to use a new name for these new test vectors.

What do you think about just use `kzg`?

```suggestion
                runner_name='kzg',
```